genai-perf/genai_perf/metrics/telemetry_metrics.py

```diff
@@ -64,11 +72,14 @@ def __init__(
         self.gpu_memory_used = gpu_memory_used
 
     def update_metrics(self, measurement_data: dict) -> None:
-        """Update the metrics with new measurement data"""
         for metric in self.TELEMETRY_METRICS:
             metric_key = metric.name
             if metric_key in measurement_data:
-                getattr(self, metric_key).append(measurement_data[metric_key])
+                metric_data = measurement_data[metric_key]
+                for gpu_name, values in metric_data.items():
+                    if gpu_name not in getattr(self, metric_key):
+                        getattr(self, metric_key)[gpu_name] = []
+                    getattr(self, metric_key)[gpu_name].extend(values)
```

Comment:
if you set `defaultdict(list)` to store the metrics, then I think you can simply do 
```suggestion
                    getattr(self, metric_key)[gpu_name].extend(values)
```

Reply:
I modified.